src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -1139,6 +1140,289 @@ public static IResourceBuilder<T> WithCommand<T>(
+            confirmationMessage: confirmationMessage,
+            iconName: iconName,
+            iconVariant: iconVariant,
+            isHighlighted: isHighlighted);
+
+    /// <summary>
+    /// Adds a command to the resource that when invoked sends an HTTP request to the specified endpoint and path.
+    /// </summary>
+    /// <typeparam name="TResource">The type of the resource.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="path">The path to send the request to when the command is invoked.</param>
+    /// <param name="displayName">The display name visible in UI.</param>
+    /// <param name="endpointSelector">A callback that selects the HTTP endpoint to send the request to when the command is invoked.</param>
+    /// <param name="method">The HTTP method to use when sending the request. Defaults to <c>POST</c>.</param>
+    /// <param name="httpClientName">The name of the HTTP client to use when creating it via <see cref="IHttpClientFactory.CreateClient(string)"/>.</param>
+    /// <param name="configureRequest">A callback to be invoked to configure the request before it is sent.</param>
+    /// <param name="getCommandResult">A callback to be invoked after the response is received to determine the result of the command invocation.</param>
+    /// <param name="commandName">The name of command. The name uniquely identifies the command.</param>
+    /// <param name="displayDescription">
+    /// Optional description of the command, to be shown in the UI.
+    /// Could be used as a tooltip. May be localized.
+    /// </param>
+    /// <param name="confirmationMessage">
+    /// When a confirmation message is specified, the UI will prompt with an OK/Cancel dialog
+    /// and the confirmation message before starting the command.
+    /// </param>
+    /// <param name="iconName">The icon name for the command. The name should be a valid FluentUI icon name from <see href="https://aka.ms/fluentui-system-icons"/></param>
+    /// <param name="iconVariant">The icon variant.</param>
+    /// <param name="isHighlighted">A flag indicating whether the command is highlighted in the UI.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <exception cref="DistributedApplicationException"></exception>
+    /// <remarks>
+    /// <para>
+    /// The command will be added to the resource represented by <paramref name="builder"/>.
+    /// </para>
+    /// <para>
+    /// If no <paramref name="endpointSelector"/> is specified, the first HTTP endpoint found on the resource will be used.
+    /// HTTP endpoints with an <c>https</c> scheme are preferred over those with an <c>http</c> scheme. If no HTTP endpoint
+    /// is found on the resource, an exception will be thrown.
+    /// </para>
+    /// <para>
+    /// The supplied <paramref name="endpointSelector"/> may return an endpoint from a different resource to that which the command is being added to.
```

Comment:
Do we have a test for this? Returning an endpoint for a different resource.

Reply:
Yes, see https://github.com/dotnet/aspire/pull/7962/files#diff-39660fc1865df29aaf6dd429fbf374c894ffe5a8330d6660706718438a0758e5R185